### DISCUSSION TITLE
Parent control and reactive properties

### DISCUSSION THREAD

**Discussion by dsgaye** [lit/lit#4623] [Q&A] [answered]
Hi everybody,

I am in a project with lit using some controls with reactive properties, but we found that when the value change inside the child control this value is not received by the parent.

Here is a simple playground to explain it
[Playground](https://lit.dev/playground/#project=W3sibmFtZSI6ImluZGV4Lmh0bWwiLCJjb250ZW50IjoiPCFkb2N0eXBlIGh0bWw-XG48aHRtbD5cbiAgPGhlYWQ-XG4gICAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9vdGhlci1lbGVtZW50LmpzXCI-PC9zY3JpcHQ-XG4gICAgPHNjcmlwdCB0eXBlPVwibW9kdWxlXCIgc3JjPVwiLi9teS1lbGVtZW50LmpzXCI-PC9zY3JpcHQ-XG4gIDwvaGVhZD5cbiAgPGJvZHk-XG4gICAgPG15LWVsZW1lbnQ-PC9teS1lbGVtZW50PlxuICA8L2JvZHk-In0seyJuYW1lIjoicGFja2FnZS5qc29uIiwiY29udGVudCI6IntcbiAgXCJkZXBlbmRlbmNpZXNcIjoge1xuICAgIFwibGl0XCI6IFwiXjMuMC4wXCIsXG4gICAgXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjogXCJeMi4wLjBcIixcbiAgICBcImxpdC1lbGVtZW50XCI6IFwiXjQuMC4wXCIsXG4gICAgXCJsaXQtaHRtbFwiOiBcIl4zLjAuMFwiXG4gIH1cbn0ifSx7Im5hbWUiOiJteS1lbGVtZW50LnRzIiwiY29udGVudCI6ImltcG9ydCB7IExpdEVsZW1lbnQsIGNzcywgaHRtbH0gZnJvbSBcImxpdFwiO1xuaW1wb3J0IHtjdXN0b21FbGVtZW50LCBwcm9wZXJ0eX0gZnJvbSAnbGl0L2RlY29yYXRvcnMuanMnO1xuXG5AY3VzdG9tRWxlbWVudCgnbXktZWxlbWVudCcpXG5jbGFzcyBNeUVsZW1lbnQgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgc3RhdGljIHN0eWxlcyA9IGNzc2BkaXYgeyBiYWNrZ3JvdW5kOiBibHVlIH1gO1xuICBcbiAgQHByb3BlcnR5KClcbiAgc2Vjb25kcHJvcDogc3RyaW5nID0gJ1RoaXMgaXMgc2Vjb25kIGNvbnRyb2wnO1xuXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGBcbiAgICAgIDxkaXY-XG4gICAgICAgIDxwPkhlbGxvLCBJJ20gcGFyZW50IGNvbnRyb2whPC9wPlxuICAgICAgICA8c3Bhbj5TZWNvbmRQcm9wIG9uIHBhcmVudCA9PiAke3RoaXMuc2Vjb25kcHJvcH08L3NwYW4-XG4gICAgICAgIDxvdGhlci1lbGVtZW50IG15cmVhY3RpdmVwcm9wPVwiJHt0aGlzLnNlY29uZHByb3B9XCI-PC9vdGhlci1lbGVtZW50PlxuICAgICAgPC9kaXY-XG4gICAgYDtcbiAgfVxufSJ9LHsibmFtZSI6Im90aGVyLWVsZW1lbnQudHMiLCJjb250ZW50IjoiaW1wb3J0IHtodG1sLCBjc3MsIExpdEVsZW1lbnR9IGZyb20gJ2xpdCc7XG5pbXBvcnQge2N1c3RvbUVsZW1lbnQsIHByb3BlcnR5fSBmcm9tICdsaXQvZGVjb3JhdG9ycy5qcyc7XG5cbkBjdXN0b21FbGVtZW50KCdvdGhlci1lbGVtZW50JylcbmV4cG9ydCBjbGFzcyBvdGhlckVsZW1lbnQgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgc3RhdGljIHN0eWxlcyA9IGNzc2BkaXYgeyBiYWNrZ3JvdW5kOiBncmVlbiB9YDtcbiAgXG4gIEBwcm9wZXJ0eSh7YXR0cmlidXRlOiB0cnVlLCByZWZsZWN0OiB0cnVlIH0pXG4gIG15cmVhY3RpdmVwcm9wOiBzdHJpbmcgPSBcImRlZmF1bHRWYWx1ZVwiO1xuICBcbiAgY2hhbmdlVmFsdWUoKXtcbiAgICB0aGlzLm15cmVhY3RpdmVwcm9wID0gXCJwYXJlbnQgc2hvdWxkIGNoYW5nZSB3aXRoIHJlYWN0aXZlIHByb3BlcnR5IGFuZC4uLiBOT1AhXCI7XG4gIH1cbiAgXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGBcbiAgICAgIDxkaXY-XG4gICAgICAgIDxwPkhlbGxvLCBpJ20gc2Vjb25kIGNvbnRyb2whPC9wPlxuICAgICAgICA8c3Bhbj5NeSBwcm9wIGNvbnRhaW5zID0-ICR7dGhpcy5teXJlYWN0aXZlcHJvcH08L3NwYW4-XG4gICAgICAgIDxidXR0b24gQGNsaWNrPVwiJHt0aGlzLmNoYW5nZVZhbHVlfVwiPkNsaWNrIHRvIGJ1ZzwvYnV0dG9uPlxuICAgICAgPC9kaXY-XG4gICAgYDtcbiAgfVxufSJ9XQ)

Is something wrong in the code or is a bug?

We need to be able to get de value of the children control from the parent without addintional functions while they are in diferents files as in the playground.

Thanks a lot for your help and support.

**Comment by sorvell** [comment, marked as answer]
This behavior is not automatic by design. Instead when you want to sync a property between a child and a parent, the child should dispatch an event that the parent handles to set its property. For additional background see the [Events docs](https://lit.dev/docs/components/events/) and [Event communication between web components](https://www.youtube.com/watch?v=T9mxtnoy9Qw). 

Here's an [updated example](https://lit.dev/playground/#gist=5fa53fa47e190e4482b073d1dd2137ef). There are 2 changes:
1. On the child element, the click event that sets the property fires a `change` event. The name of this can be whatever you want. The parent then handles this event `@change` and sets its property to the child property. 
2. The child part is bound via the property and not the attribute (add the `.` here `.myreactiveprop`). This isn't strictly necessary but it's typically better to bind properties when possible since this always works and attributes must be serializable as strings, which can be limiting for some data.

**Comment by dsgaye** [comment]
Thank you for your fast answer!
Do you know if in the future will be implemented a versión of Lit that allow to be able to do it in a similar way as my example?
